src/enclave/Enclave/ExpressionEvaluation.h

```diff
@@ -107,7 +107,8 @@ template<typename TuixExpr, template<typename T> class Operation>
 flatbuffers::Offset<tuix::Field> eval_binary_comparison(
   flatbuffers::FlatBufferBuilder &builder,
   const tuix::Field *left,
-  const tuix::Field *right) {
+  const tuix::Field *right,
+  bool is_null_asc = true) {
```

Comment:
It would be good to mention this `is_null_asc` parameter in the method description, particularly because it does something surprising: it uses the underlying value of a NULL field to return extra information.

Reply:
Also, the parameter name is a little confusing. How about `nulls_first`? This follows the Spark SQL [NullOrdering](https://github.com/apache/spark/blob/0b3bb45b893005d73afb15e4904ac6440c87811c/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/SortOrder.scala#L47) naming.

Reply:
It would also be good to call out the surprising use of this method at the call site where it occurs ([`FlatbuffersSortOrderEvaluator::less_than()`](https://github.com/mc2-project/opaque/blob/e890e2aae4751a215ce43cd47c1168b67077353c/src/enclave/Enclave/ExpressionEvaluation.h#L1137)). Something like "We ignore the result's NULL flag and use only its underlying value for the comparison. `eval_binary_comparison()` uses this underlying value to pass the desired information."

Reply:
Improved comments.